tests/main/gadget-update-pc/task.yaml

```diff
@@ -0,0 +1,164 @@
+summary: Exercise a gadget update on a PC
+
+environment:
+    BLOB_DIR: $(pwd)/fake-store-blobdir
+    # snap-id of 'pc' gadget snap
+    PC_SNAP_ID: UqFziVZDHLSyO3TqSWgNBoAdHbLI4dAH
+    START_REVISION: 1000
+
+systems: [ubuntu-core-*]
+
+prepare: |
+    if ! test -d /snap/pc; then
+        echo "This test needs a host using 'pc' gadget snap"
+        exit 0
+    fi
+
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    snap ack "$TESTSLIB/assertions/testrootorg-store.account-key"
+
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    setup_fake_store "$BLOB_DIR"
+
+    cp /var/lib/snapd/snaps/pc_*.snap gadget.snap
+    unsquashfs -d pc-snap gadget.snap
+
+    # gadget YAMLs should be identical, otherwise the test needs to be updated
+    diff -up pc-gadget.yaml pc-snap/meta/gadget.yaml
+
+    # prepare a vanilla version
+    sed -i -e 's/^version: \(.*\)/version: \1-1/' pc-snap/meta/snap.yaml
+    # pack it
+    snap pack pc-snap --filename=pc_x1.snap
+
+    cat <<EOF > decl-headers.json
+    {"snap-id": "$PC_SNAP_ID"}
+    EOF
+    cat <<EOF > rev-headers.json
+    {"snap-id": "$PC_SNAP_ID", "snap-revision": "$START_REVISION"}
+    EOF
+
+    new_snap_declaration "$BLOB_DIR" pc_x1.snap --snap-decl-json decl-headers.json
+    new_snap_revision "$BLOB_DIR" pc_x1.snap --snap-rev-json rev-headers.json
+
+    # prepare first update
+    echo 'this is foo' > foo-2.img
```

Comment:
Where is foo-2.img used ? The foo-x2.img below is used but afaict this one is not?